packages/advanced-logic/src/extensions/payment-network/any-to-native.ts

```diff
@@ -0,0 +1,60 @@
+import { FeeReferenceBasedPaymentNetwork } from './fee-reference-based';
+import { ExtensionTypes, RequestLogicTypes } from '@requestnetwork/types';
+import { InvalidPaymentAddressError } from './address-based';
+
+export default abstract class AnyToNativeTokenPaymentNetwork extends FeeReferenceBasedPaymentNetwork {
+  public constructor(
+    extensionId: ExtensionTypes.ID,
+    currentVersion: string,
+    supportedNetworks: string[],
+  ) {
+    super(extensionId, currentVersion, supportedNetworks, RequestLogicTypes.CURRENCY.ETH);
+  }
+
+  public createCreationAction(
+    creationParameters: ExtensionTypes.PnAnyToAnyConversion.ICreationParameters,
+  ): ExtensionTypes.IAction<ExtensionTypes.PnAnyToAnyConversion.ICreationParameters> {
+    const network = creationParameters.network;
+    if (!network) {
+      throw Error('network is required');
+    }
+
+    this.throwIfInvalidNetwork(network, this.supportedNetworks);
```

Comment:
If I remember correctly, I did not integrate it to the `throwIfInvalidNetwork` function because then the `network` variable would still be considered as possibly undefined which raised errors.
I've made an update so the check returns the `network` property as a string if there's no error thrown.

Reply:
To resolve this you could do:
```ts
  protected throwIfInvalidNetwork(network?: string): asserts network is string {
    if (!network) {
      throw Error('network is required');
    }
    if (network && this.supportedNetworks && !this.supportedNetworks.includes(network)) {
      throw new UnsupportedNetworkError(network, this.supportedNetworks);
    }
  }
```

Reply:
@alexandre-abrioux nice, I did not know about this TS feature, thanks !